src/core/dev_api/bound_evaluation_util.hpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
+
+namespace ov {
+bool could_propagate(const Output<Node>& output, std::vector<Node*>& order);
+
+/// \brief Checks if all the elements of the bound Tensor are positive
+bool tensor_is_positive(const Tensor& bound);
+
+/// \brief Estimates upper bound for node output tensors using only upper bounds of the nodes
+/// inputs.
+/// \param node Operation to be performed
+/// \param output_values Vector of Tensors representing resulting upper value estimations
+/// \return boolean status if value evaluation was successful.
+bool default_upper_bound_evaluator(const Node* node, TensorVector& output_values);
+
+/// \brief Estimates lower bound for node output tensors using only lower bounds of the nodes
+/// inputs.
+/// \param node Operation to be performed
+/// \param output_values Vector of Tensors representing resulting lower value estimations
+/// \return boolean status if value evaluation was successful.
+bool default_lower_bound_evaluator(const Node* node, TensorVector& output_values);
```

Comment:
If it is a part of API and we use it in other components it should be exported.
If we use this API only inside core, I propose to remove these methods from the developer API and move to `src` folder.

Reply:
Moved not required functions to `src` folder and add export macro for those which are in dev API.
Extract sources of bound evaluation functions to separate file.